jax/numpy/lax_numpy.py

```diff
@@ -1768,16 +1760,23 @@ def identity(n, dtype=None):
 @_wraps(onp.arange)
 def arange(start, stop=None, step=None, dtype=None):
   lax._check_user_dtype_supported(dtype, "arange")
-  # If called like np.arange(N), we create a lazy lax._IotaConstant.
   if stop is None and step is None:
     dtype = dtype or _dtype(start)
-    if issubdtype(dtype, onp.integer):
-      return lax.iota(dtype, start)  # avoids materializing
-
-  # Fall back to instantiating an ndarray in host memory
-  dtype = dtype or result_type(
-    *(x for x in (start, stop, step) if x is not None))
-  return onp.arange(start, stop=stop, step=step, dtype=dtype)
+    return lax.iota(dtype, start)  # avoids materializing
+  elif step is None or onp.issubdtype(_dtype(step), onp.integer):
```

Comment:
Use `issubdtype`? I removed all uses of `onp.issubdtype` from JAX, so I would prefer we don't reintroduce them. (They are problematic because they no longer do the same thing in some cases, although I admit this is not one of them.)

Reply:
Amazing catch, thank you! Done.